src/esssans/i_of_q.py

```diff
@@ -236,87 +217,43 @@ def _to_q_bins(q_bins: Union[int, sc.Variable]) -> Dict[str, Union[int, sc.Varia
 
 
 def _events_merge_spectra(
-    data_q: sc.DataArray,
-    q_bins: Union[int, sc.Variable],
-    dims_to_reduce: List[str],
-    wavelength_bands: sc.Variable,
+    data_q: sc.DataArray, q_bins: Union[int, sc.Variable], dims_to_reduce: List[str]
 ) -> sc.DataArray:
     """
     Merge spectra of event data
     """
-    q_all_pixels = data_q.bins.concat(dims_to_reduce)
     edges = _to_q_bins(q_bins)
-    q_binned = q_all_pixels.bin(**edges)
-    dim = 'wavelength'
-    wav_binned = q_binned.bin({dim: sc.sort(wavelength_bands.flatten(to=dim), dim)})
-    # At this point we kind of already have what we need, would be cheapest to just
-    # return, if follow up providers can work with the result.
-    # Otherwise we need to duplicate events:
-    sections = []
-    for bounds in sc.collapse(wavelength_bands, keep=dim).values():
-        # The extra concat can probably be avoided if we insert some dummy edges for
-        # first and last band, but we would need to know how many edges to insert, as
-        # the bands can be very wide and overlap by more than one bin.
-        sections.append(wav_binned[dim, bounds[0] : bounds[1]].bins.concat(dim))
-    band_dim = (set(wavelength_bands.dims) - {'wavelength'}).pop()
-    out = sc.concat(sections, band_dim)
-    out.coords[dim] = wavelength_bands
-    return out
+    q_all_pixels = data_q.bins.concat(dims_to_reduce)
```

Comment:
Now that the body of this function is so small, and the one for the dense merging is also much reduced, would it make more sense to just inline the code in `merge_spectra`?

I don't mind, it's a matter of taste. It may still be more readable as it is now.

Reply:
Done